threads/src/main/java/org/jboss/as/threads/ThreadFactoryResourceDefinition.java

```diff
@@ -46,13 +49,19 @@ public ThreadFactoryResourceDefinition() {
     }
 
     public ThreadFactoryResourceDefinition(String type) {
-        super(PathElement.pathElement(type),
+        this(PathElement.pathElement(type), RuntimeCapability.Builder.of("org.wildfly.threads." + type , true, ThreadFactory.class).build());
```

Comment:
should `type` be an open-ended string or rather an enum that list all the supported types.

Reply:
There was no previous constraint on type, so an open ended string is acceptable.